samples/core/output_a_directory/output_a_directory_test.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2021 The Kubeflow Authors
```

Comment:
I wonder whether it's possible to not need these test launchers for every sample.

Perhaps we could move all of them to the sample test dir.

Reply:
Can you provide a sample?
I just realized local import might not be possible in legacy e2e tests https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/5473/kubeflow-pipeline-e2e-test/1382342847985881088#1:build-log.txt%3A5624.

I need to use an url import here.

Reply:
I'd suggest keeping them here, they are useful for several reasons:

* allow custom verification logic for each test -- especially verifying the MLMD data structure (I plan to add some examples, and I believe they are the true value for a pipeline test, verifying a pipeline can succeed is not verifying much)
* allow arbitrary number of testcases with different setup (test v1_legacy, test v2_compatible, test different arguments)
* keep test next to code

Reply:
I think we might want to have some separation between what we show to the users and the internal test infra machinery.

It's OK to have multiple test cases for a single samples, but I'm not sure we should show the users with that test infra code.

Reply:
Created https://github.com/kubeflow/pipelines/issues/5506 to track further discussion.